services/rollout-service/pkg/service/service.go

```diff
@@ -102,23 +101,14 @@ func (e *ArgoEventConsumer) ConsumeEvents(ctx context.Context) error {
 					logger.FromContext(ctx).Sugar().Warnf("argo apps channel at full capacity of %d. Discarding event: %v", cap(e.ArgoAppProcessor.ArgoApps), ev)
 				}
 
-				if e.DDMetrics != nil { //If DD is enabled, send metrics
+				if e.DDMetrics != nil && e.ArgoAppProcessor.ArgoAppsMetricsEnabled { //If DD is enabled, send metrics
```

Comment:
Please add an early exit (probably in the start of this function) if `e.ArgoAppProcessor == nil`